test/test_autograd.py

```diff
@@ -3333,6 +3333,343 @@ def test_checkpointing(self):
         mean_combined = torch.stack(feat_combined).mean()
         mean_combined.backward()
 
+    def test_autograd_views_codegen(self):
+        # This is not necessarily the absolute correct behavior, but this is the current
+        # one. This test is here to make sure that any change to this behavior is detected
+        # and not silent. The TODOs below mark the places with unexpected behavior.
+        # Note that any change in these test will be BC-breaking and should be done carefully.
+
+        # This test checks the behavior of two codegen functions (view_as and unbind)
+        # with respect to view tracking and inplace operation on the output.
+
+        def run_test(grad_mode, requires_grad, is_view, should_raise_tuple):
+            def get_view_as():
+                inp = torch.rand(2, requires_grad=requires_grad).clone()
+                with torch.set_grad_enabled(grad_mode):
+                    return inp.view_as(inp)
+
+
+            def get_unbind():
+                inp = torch.rand(2, requires_grad=requires_grad).clone()
+                with torch.set_grad_enabled(grad_mode):
+                    return inp.unbind()
+
+            def maybe_check_raise(fn, should_raise):
+                self.assertTrue(should_raise is None or isinstance(should_raise, str))
+                if should_raise is not None:
+                    with self.assertRaisesRegex(RuntimeError, should_raise):
+                        fn()
+                else:
+                    fn()
+
+            # Are they differentiable views?
+            out = get_view_as()
+            self.assertTrue(out._is_view() == is_view)
+
+            out = get_unbind()
+            self.assertTrue(out[0]._is_view() == is_view)
+            self.assertTrue(out[1]._is_view() == is_view)
+
+            # Are inplace allowed?
+            out = get_view_as()
```

Comment:
I don't understand, why do you need to allocate fresh variables for the inplace test? The code here would be a lot more direct if you just inlined the `view_as` and `get_unbind` implementations; they seem to be factored into functions so you can call them twice but the second call seems completely unnecessary.

Reply:
Yes I'll clean this up.